decisions/0003-files-in-JSON-output.md

```diff
@@ -121,27 +144,183 @@ lost
+offer a mechanism that enables these entities to be able to integrate with other
+systems
+
+### Whether to send files to the receiver or let them know they should request them from Form Builder
+
+The latter is more likely to be reliable and scalable, so proceeding on this
+basis this excludes options 1 & 4
+
+### Whether to encrypt the files
+
+Presumably with the PSK used to encrypt the JSON payload, since they will
+previously be encrypted for the submitter which has finished with them by this
+stage in one or both of:
+
+- At rest in S3
+- In transit to the receiver
+
+### How long to keep files retrievable for
+
+Whether directly available from S3 or through a proxy application by the
+receiver. In some use cases this may involve human intervention to get files
+into the CMS.
+
+### Whether to use S3 native signed URLs or generate and manage our own signed URLs
+
+If the latter, whether to only generate single-use signed URLs on request
+(with those requests authenticated as being from the receiver) or whether to use
+time-limited URLs, possibly with ability to revoke them on callback confirming
+that the receiver has successfully retrieved and stored the file.
 
-This create a lightweight JSON payload which reduces processing required by both
-parties.
+### Issues concerning encryption:
 
-The recipient is only required to accept a small JSON payload and can
-asynchronously pickup the files at a later point in time. To prevent files
-exposed, signed URLs should be short lived so if they were leaked the
-information should no longer be accessible.
+- If we don't encrypt the file for the receiver in S3, or make it available at a
+URL unencrypted, then we need to be much more careful about bucket
+misconfiguration, short lived URLs, revokable URLs, alerting on reuse of URLs,
+attempts to traverse/enumerate bucket contents should be considered to prevent
+attempts to gain access from unintended audiences.
 
-As we control these URLs it is possible to also generate one time use URLs.
-After we know the URL has been called, the token can be revoked.
+- If we do encrypt the files in S3, we don't need to worry so much about
+accidental misconfiguration of the bucket or attempts to traverse/enumerate it -
+even if we don't have a proxy application in front of it
 
-As the files are encrypted we must sit a proxy in front of S3 to handle this.
-However this will give Form Builder full control of access to files stored.
+- If we encrypt the files in S3 and also serve them encrypted, then we don't
+need to worry as much about bucket-level access to the files.
+
+- A proxy application is needed if the adapter does not have the responsibility
+of decrypting the files. A proxy would not be needed if the responsibility of
+decryption is moved to the adapter. However some kind of pre-shared key
+mechanism would be needed.
+
+- This also follows the principle of all user-submitted data in Form Builder
```

Comment:
This and the following two points refer specifically to the case of encrypting files for the receiver at rest and in transit - that's not clear from the formatting here.

Reply:
I've added to the end of the paragraph hopefully that makes it clear